src/pages/lessons/[slug].tsx

```diff
@@ -242,22 +211,51 @@ const Lesson: FunctionComponent<LessonProps> = ({initialLesson}) => {
 
         <div
           className="relative overflow-hidden bg-gray-100"
-          style={{paddingTop: '56.25%'}}
+          css={{paddingTop: '56.25%'}}
         >
-          {playerState.value === 'playing' && (
-            <EggheadPlayer
-              ref={playerRef}
-              className="absolute top-0 left-0 w-full h-full"
-              hls_url={hls_url}
-              dash_url={dash_url}
-              width="100%"
-              height="auto"
-              pip="true"
-              controls
-              subtitlesUrl={get(lesson, 'subtitles_url')}
-            />
-          )}
+          <div className="absolute w-full h-full top-0 left-0">
+            {playerVisible && (
+              <EggheadPlayer
+                ref={playerRef}
+                hls_url={hls_url}
+                dash_url={dash_url}
+                width="100%"
+                height="auto"
+                pip="true"
+                controls
+                onPlay={() => send('PLAY')}
+                onPause={() => send('PAUSE')}
+                onEnded={() => send('COMPLETE')}
+                subtitlesUrl={get(lesson, 'subtitles_url')}
+              />
+            )}
+
+            {playerState.value === 'subscribe' && (
+              <div className="flex justify-center items-center h-full">
+                <Link href="/pricing">
+                  <a>Get Access to This Video</a>
+                </Link>
+              </div>
+            )}
+
+            {playerState.value === 'showingNext' && (
+              <div className="flex justify-center items-center h-full">
+                <NextResourceButton
+                  path={nextUpPath}
+                  onClick={() => send('LOAD')}
+                >
+                  Load the Next Video
+                </NextResourceButton>
+              </div>
+            )}
+          </div>
         </div>
+        <PlayerControls
+          handlerDownload={() => console.log('handlerDownload')}
+          isPro={true}
+        >
+          <NextResourceButton path={nextUpPath} onClick={() => send('LOAD')} />
```

Comment:
here's an example of using component composition

https://www.robinwieruch.de/react-component-composition

Instead of burying the next button in this component we can pass it in as a child. This means we don't have to pass in the `send` method to change state.